env/fs_posix.cc

```diff
@@ -554,23 +554,31 @@ class PosixFileSystem : public FileSystem {
                    std::shared_ptr<Logger>* result,
                    IODebugContext* /*dbg*/) override {
     FILE* f;
+    int fd;
     {
       IOSTATS_TIMER_GUARD(open_nanos);
-      f = fopen(fname.c_str(),
-                "w"
+      fd = open(fname.c_str(), O_WRONLY | O_CREAT | O_TRUNC | O_CLOEXEC, 0644);
```

Comment:
This should be open(fname.c_str(), cloexec_flags(O_WRONLY | O_CREAT | O_TRUNC, nullptr), GetDBFileMode(allow_non_owner_access_));